server/events/vcs/github_client.go

```diff
@@ -273,7 +274,25 @@ func (g *GithubClient) PullIsMergeable(repo models.Repo, pull models.PullRequest
 
 // GetPullRequest returns the pull request.
 func (g *GithubClient) GetPullRequest(repo models.Repo, num int) (*github.PullRequest, error) {
-	pull, _, err := g.client.PullRequests.Get(g.ctx, repo.Owner, repo.Name, num)
+	var err error
+	var pull *github.PullRequest
+
+	// GitHub has started to return 404's here (#1019) even after they send the webhook.
+	// They've got some eventual consistency issues going on so we're just going
+	// to retry up to 3 times with a 1s sleep.
+	numRetries := 3
+	retryDelay := 1 * time.Second
```

Comment:
Thank you for implementing this retries! 🤟 

could you also please implement exponential backoff to space out the retries? 
  1. on first retry you sleep for 1 sec
  1. on second retry you sleep for 2 sec
  1. on third retry you sleep for 4 sec.